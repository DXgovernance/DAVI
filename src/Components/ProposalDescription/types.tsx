interface metadata {
  description: string;
}

export interface ProposalDescriptionProps {
  t: any;
  metadata?: metadata;
  error?: any;
}
