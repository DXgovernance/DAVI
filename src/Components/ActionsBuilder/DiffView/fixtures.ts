export const mockCodeOriginal = `
  {
    "colors": {
      "text": "red",
      "primary": "#295FF4",
      "secondary": "#181B1B",
      "red": "#E75C5C",
      "orange": "#D17A06",
      "proposalText": {
        "lightGrey": "#BDC0C7",
        "grey": "orange"
      },
    }
  }
`;

export const mockCodeUpdated = `
  {
    "colors": {
      "text": "#fff",
      "primary": "#295FF4",
      "secondary": "#181B1B",
      "red": "#E75C5C",
      "orange": "#D17A06",
      "proposalText": {
        "grey": "#A1A6B0",
        "lightGrey": "#BDC0C7"
      },
    }
  }
`;
