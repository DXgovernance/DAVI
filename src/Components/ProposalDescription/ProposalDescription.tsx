import { ProposalDescriptionWrapper } from './ProposalDescription.styled';
import { ProposalDescriptionProps } from './types';
import Markdown from 'markdown-to-jsx';
import { Loading } from 'Components/Primitives/Loading';
import { useTranslation } from 'react-i18next';

const ProposalDescription: React.FC<ProposalDescriptionProps> = ({
  metadata,
  error,
}) => {
  const { t } = useTranslation();

  if (error) {
    return (
      <ProposalDescriptionWrapper>
        {t('errorMessage.proposalDescriptionError')}
      </ProposalDescriptionWrapper>
    );
  }

  return (
    <ProposalDescriptionWrapper>
      {metadata?.description ? (
        <Markdown>{metadata.description}</Markdown>
      ) : (
        <Loading loading text skeletonProps={{ width: '100%' }} />
      )}
    </ProposalDescriptionWrapper>
  );
};

export default ProposalDescription;
