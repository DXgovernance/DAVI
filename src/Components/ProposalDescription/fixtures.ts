import { ProposalDescriptionProps } from './types';

export const withMetadata: ProposalDescriptionProps = {
  metadata: {
    description: 'This is the description of the proposal',
  },
  t: key => key,
};

export const noMetadata: ProposalDescriptionProps = {
  t: key => key,
};

export const errorMetadata: ProposalDescriptionProps = {
  error: 'error',
  t: key => key,
};
