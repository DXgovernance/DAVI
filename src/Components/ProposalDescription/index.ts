export { default as ProposalDescription } from './ProposalDescription';

