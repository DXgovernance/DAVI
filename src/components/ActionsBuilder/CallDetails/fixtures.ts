import {
  ApproveSendTokens,
  DecodedCall,
  SupportedAction,
} from 'components/ActionsBuilder/types';
import ERC20 from 'contracts/ERC20.json';
import { BigNumber, utils } from 'ethers';

const ERC20Contract = new utils.Interface(ERC20.abi);

export const emptyDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '',
  value: BigNumber.from(0),
  args: {
    _to: '',
    _value: BigNumber.from(0),
  },
};

export const decodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '0x95a223299319022a842d0dfe4851c145a2f615b9',
  value: BigNumber.from(0),
  args: {
    _to: '0x0000000000000000000000000000000000000000',
    _value: BigNumber.from(1000),
  },
};

export const approvetokenSpendingMock: ApproveSendTokens = {
  callType: SupportedAction.GENERIC_CALL,
  from: '0x0000000000000000000000000000000000000000',
  to: '0x0000000000000000000000000000000000000000',
  value: BigNumber.from(0),
  function: null,
  args: {},
  amount: BigNumber.from(1000),
  token: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
};
