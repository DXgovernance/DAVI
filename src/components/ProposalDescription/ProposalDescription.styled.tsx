import styled from 'styled-components';

export const ProposalDescriptionWrapper = styled.div`
  margin: 1.5rem 0;
  line-height: 1.5;
  font-size: 16px;
  word-break: break-word;
  text-align: left;
  color: ${({ theme }) => theme.colors.grey2};
`;
