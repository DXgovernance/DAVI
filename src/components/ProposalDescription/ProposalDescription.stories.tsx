import { ComponentMeta, ComponentStory } from '@storybook/react';
import { useTranslation } from 'react-i18next';
import { ProposalDescription } from './ProposalDescription';
import { withMetadata, noMetadata, errorMetadata } from './fixtures';

export default {
  title: 'ProposalDescription',
  component: ProposalDescription,
  argTypes: {
    t: {
      description: 'Translation library object',
    },
    metadata: {
      description:
        'Proposal metadata, used to populate the description of the proposal',
    },
    errorMetadata: {
      description: 'Error element populated if the metadata fetch fails',
    },
  },
} as ComponentMeta<typeof ProposalDescription>;

const TranslationWrapper = ({ args }) => {
  const { t } = useTranslation();
  args.t = t;

  return <ProposalDescription {...args} />;
};

const Template: ComponentStory<typeof ProposalDescription> = args => (
  <TranslationWrapper args={args} />
);

export const Simple = Template.bind({});
Simple.args = withMetadata;

export const Loading = Template.bind({});
Loading.args = noMetadata;

export const Error = Template.bind({});
Error.args = errorMetadata;
