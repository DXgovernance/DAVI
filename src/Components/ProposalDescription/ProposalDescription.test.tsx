import { render } from 'utils/tests';
import ProposalDescription from './ProposalDescription';
import { withMetadata, noMetadata, errorMetadata } from './fixtures';

jest.mock('contexts/index', () => jest.fn());

describe('ProposalDescription', () => {
  it('Should render with description', () => {
    const { container } = render(<ProposalDescription {...withMetadata} />);
    expect(container).toMatchSnapshot();
  });

  it('Should render loading if it has no metadata', () => {
    const { container } = render(<ProposalDescription {...noMetadata} />);
    expect(container).toMatchSnapshot();
  });

  it('Should render an error', () => {
    const { container } = render(<ProposalDescription {...errorMetadata} />);
    expect(container).toMatchSnapshot();
  });
});
