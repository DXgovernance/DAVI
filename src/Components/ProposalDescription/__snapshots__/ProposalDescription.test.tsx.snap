// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProposalDescription Should render an error 1`] = `
<div>
   
  <div
    class="sc-bdfBQB kCzFam"
  >
    errorMessage.proposalDescriptionError
  </div>
   
</div>
`;

exports[`ProposalDescription Should render loading if it has no metadata 1`] = `
<div>
   
  <div
    class="sc-bdfBQB kCzFam"
  >
    <div
      class="sc-gsTEea"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="width: 100%; --base-color: #333; --highlight-color: #555;"
        >
          â€Œ
        </span>
        <br />
      </span>
    </div>
  </div>
   
</div>
`;

exports[`ProposalDescription Should render with description 1`] = `
<div>
   
  <div
    class="sc-bdfBQB kCzFam"
  >
    <span>
      This is the description of the proposal
    </span>
  </div>
   
</div>
`;
